### PR TITLE
Use taxable address when adding/removing coupons

### LINKED ISSUE #22643: WooCommerce Tax Calculation
**BUG**
Whenever an admin user creates an order from admin panel, in the store where the TAX calculations and use of coupons is active. There are certain standard tax rates being applicable, based on **customer shipping address**.
Now when the admin creates an order, adds an item and applies a coupon, the tax is being calculated based in store address, and not on customer's shipping address.
The complete bug is explained in the screencast attached.

**To Reproduce**
To reproduct the issue, kindly watch the screencast attached.

**Expected behavior**
I expect that when the tax calculations is set to be calculated based on customer's shipping address, then it should be calculated on customer's shipping address, irrelevant of the store address.

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] I can reproduce this bug consistently watching the screencast attached above.

**WordPress Environment**
Version - 5.3

**WooCommerce Environment**
Version - 5.3.4
System status - [system-status.txt](https://github.com/woocommerce/woocommerce/files/2826899/system-status.txt)

Bug description - [woocommerce_tax_calculation_issue.zip](https://github.com/woocommerce/woocommerce/files/2826932/woocommerce_tax_calculation_issue.zip)

I hope you'll understand the issue. If I'm understanding WooCommerce in a wrong manner, please do let me know.

Thank you!

### PR DESCRIPTION
**Pull request by mikejolley**
See #22643 for video showing issue/how to test.

Fixes #22643

Like when totals are recalculated, passes the unsaved address through to the add/remove coupon endpoints so when taxes are calculated they use customer address, not store address.

> Use correct address for taxes when applying/removing coupons.

### CODE REVIEW

File: includes/class-wc-ajax.php [hunk 1]
```diff
@@ -1076,14 +1076,24 @@ public static function add_coupon_discount() {
...
-			$order_id = absint( $_POST['order_id'] );
-			$order    = wc_get_order( $order_id );
-			$result   = $order->apply_coupon( wc_clean( $_POST['coupon'] ) );
+			$order_id           = absint( $_POST['order_id'] );
+			$order              = wc_get_order( $order_id );
+			$calculate_tax_args = array(
+				'country'  => strtoupper( wc_clean( $_POST['country'] ) ),
+				'state'    => strtoupper( wc_clean( $_POST['state'] ) ),
+				'postcode' => strtoupper( wc_clean( $_POST['postcode'] ) ),
+				'city'     => strtoupper( wc_clean( $_POST['city'] ) ),
```
Comment (peterfabian) [inline includes/class-wc-ajax.php]: Is there a reason not to unslash these? I can't match N'Djamena city specific tax as it gets sent to db as 
```
( locations2.location_type = 'city' AND locations2.location_code = 'N\\\'DJAMENA' )
```

(Same goes for the normal Recalculate button I think)

**Review by peterfabian — COMMENTED** [review]
This works fine except when the tax is city-specific and city has quote in the name.

File: includes/class-wc-ajax.php (hunk 1 above)
Comment (mikejolley) [reply includes/class-wc-ajax.php]: This was resolved with the PHPCS merge.

**Review by peterfabian — APPROVED** [review]
Looks good now.

**Pull request merged**